README.md

```diff
@@ -125,16 +125,47 @@ The [`test/setup-ca-kubernetes.sh`](test/setup-ca-kubernetes.sh) script shows ho
+--- | --- | --- | ---
+Persistent | supported | supported | topology aware scheduling<sup>1</sup>
+Ephemeral | [in design](https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/20190122-csi-inline-volumes.md#proposal) | in design | topology aware scheduling<sup>1</sup>, resource constraints<sup>2</sup>
+Cache | supported | supported | topology aware scheduling<sup>1</sup>
+
+<sup>1 </sup>[Topology aware scheduling](https://github.com/kubernetes/enhancements/issues/490)
+ensures that an application runs on a node where the volume was created. For CSI-based drivers like PMEM-CSI, Kubernetes >= 1.13 is needed. On older Kubernetes releases, pods must be scheduled manually onto the right node(s).
+
+<sup>2 </sup>The upstream design for ephemeral volumes currently does not take [resource constraints](https://github.com/kubernetes/enhancements/pull/716#discussion_r250536632) into account. If an application gets scheduled onto a node and then creating the ephemeral volume on that node fails, the application on the node cannot start until resources become available.
+
+#### Usage on Kubernetes
+
+Kubernetes cluster administrators can expose above mentioned [volume persistency types](#volume-persistency) to applications using [`StorageClass Parameters`](https://kubernetes.io/docs/concepts/storage/storage-classes/#parameters). An optional `persistencyModel` parameter differentiates how the provisioned volume can be used.
+
+* if no `persistencyModel` parameter specified in `StorageClass` then it is treated as normal Kubernetes persistent volume. In this case PMEM-CSI creates PMEM volume on a node and the application that claims to use this volume is supposed to schedule on this node by Kubernetes. Choosing of node is depend on StorageClass `volumeBindingMode`. In case of `volumeBindingMode: Immediate` PMEM-CSI chooses a node randomly, and in case of `volumeBindingMode: WaitForFirstConsumer` Kubernetes first chooses a node for scheduling the application, and PMEM-CSI creates the volume on that node. Applications which claim a normal persistent volume has to use `ReadOnlyOnce` access mode in its `accessModes` list. This [diagram](/docs/images/sequence/pmem-csi-persistent-sequence-diagram.png) illustrates how a normal persistent volume gets provisioned in Kubernetes using PMEM-CSI driver.
+
+* `persistencyModel: cache`</br>
+Volumes of this type shall be used in combination with `volumeBindingMode: Immediate`. In this case PMEM-CSI creates set of PMEM volumes each volume on different node. The number of PMEM volumes to create can be specified by `cacheSize` StorageClass parameter. Applications which claim a `cache` volume can use `ReadWriteMany` in its `accessModes` list. Check with provided [cache StorageClass](deploy/kubernetes-1.13/pmem-storageclass-cache.yaml) example. This [diagram](/docs/images/sequence/pmem-csi-cache-sequence-diagram.png) illustrates how a cache volume gets provisioned in Kubernetes using PMEM-CSI driver.
```

Comment:
See other comment about `<br>`.

Commas and "a": "In this case, PMEM-CSI creates a set of PMEM volumes, each volume on a different node."

"can" or "must" applications use `ReadWriteMany`?

Actually, is it the application or the storage class which chooses the access mode? Or both? The example app doesn't specify it. It would be good to have example apps+pvc for both models in `deploy` and also adapt the "make start" such that it deploys both storage classes and shows the commands for using both.


Can you add a note about what happens when multiple pods run on the same node ("they share the same volume") and suggest ways to spread out the pods if that isn't desired (not sure, but daemon set might work, or a replicaset with inter-pod anti-affinity).

Reply:
> "can" or "must" applications use `ReadWriteMany`?
> 
> Actually, is it the application or the storage class which chooses the access mode? Or both?

Application(PVC) requests the volume access modes. Application can request `ReadWriteMany` volume in case of cache volume, but its not must, if app wants to use only on a single node still it can request `ReadOnlyOnce`

Reply:
> Commas and "a": "In this case, PMEM-CSI creates a set of PMEM volumes, each volume on a different node."

Done

Reply:
> Can you add a note about what happens when multiple pods run on the same node ("they share the same volume") and suggest ways to spread out the pods if that isn't desired (not sure, but daemon set might work, or a replicaset with inter-pod anti-affinity).

Added a note, and added  replicaset example that uses the cache volume with nodeAntiAffinity.

Reply:
> It would be good to have example apps+pvc for both models in `deploy` and also adapt the "make start" such that it deploys both storage classes and shows the commands for using both.

Done.